src/twisted/web/test/test_agent.py

```diff
@@ -3216,6 +3216,24 @@ def test_cancel(self):
         self.failureResultOf(deferred, defer.CancelledError)
         self.assertTrue(response.transport.aborting)
 
+    def test_cancelWithConnectionLost(self):
+        """
+        When cancelling the L{Deferred} returned by L{client.readBody}, the
+        CancelledError failure is sent repeatedly in response protocol. Exception
+        defer.AlreadyCalledError should be suppressed
+        """
+        response = DummyResponse()
+        deferred = client.readBody(response)
+        deferred.cancel()
```

Comment:
Maybe add a comment to explain how this test is put together
```suggestion
        # It is cancelled by the user before we get the
        # full body.
        # For example due to a timeout or some other
        # conditions.
        deferred.cancel()
```